build/nuke/Build.Steps.cs

```diff
@@ -328,10 +321,15 @@ partial class Build
                     // Major and Minor version are extracted from framework and default value of 0 is appended for patch.
                     .SetOutput(AdditionalDepsDirectory / "shared" / "Microsoft.NETCore.App" / framework.ToString().Substring(framework.ToString().Length - 3) + ".0")));
 
-            AdditionalDepsDirectory.GlobFiles("**/*.dll", "**/*.pdb", "**/*.xml", "**/*.nupkg").ForEach(DeleteFile);
+            AdditionalDepsDirectory.GlobFiles("**/*.dll", "**/*.pdb", "**/*.xml").ForEach(DeleteFile);
             AdditionalDepsDirectory.GlobFiles("**/*deps.json")
-                                   .ForEach(file =>
-                                        File.WriteAllText(file, Regex.Replace(File.ReadAllText(file), $"{Projects.AdditionalDeps}.dll", $"lib/{TargetFramework.NETCOREAPP3_1}/{Projects.AdditionalDeps}.dll")));
+                                   .ForEach(file => {
+                                       string depsJsonContent = File.ReadAllText(file);
+                                       // Remove OpenTelemetry.Instrumentation.AdditionalDeps entry from target section.
+                                       depsJsonContent = Regex.Replace(depsJsonContent, "\"OpenTelemetry(.+)AdditionalDeps.dll(.+?)},\r\n(.+?)\"", "\"", RegexOptions.IgnoreCase | RegexOptions.Singleline);
+                                       // Remove OpenTelemetry.Instrumentation.AdditionalDeps entry from library section and write to file.
+                                       File.WriteAllText(file, Regex.Replace(depsJsonContent, "\"OpenTelemetry(.+?)},\r\n(.+?)\"", "\"", RegexOptions.IgnoreCase | RegexOptions.Singleline));
```

Comment:
nit (for me it is too much happening in one line - also the suggestion would make it a little easier to debug)

```suggestion
                                      depsJsonContent = Regex.Replace(depsJsonContent, "\"OpenTelemetry(.+?)},\r\n(.+?)\"", "\"", RegexOptions.IgnoreCase | RegexOptions.Singleline);
                                      File.WriteAllText(file, depsJsonContent);
```